room/compiler-processing/src/main/java/androidx/room/compiler/processing/XElement.kt

```diff
@@ -75,3 +78,18 @@ fun XElement.isConstructor(): Boolean {
     }
     return this is XConstructorElement
 }
+
+/**
+ * Attempts to get a Javac [Element] representing the originating element for attribution
+ * when writing a file for incremental processing.
+ *
+ * Note, instead of using this directly you can instead use the [addOriginatingElement] extension
+ * functions for JavaPoet and KotlinPoet Type builders.
+ */
+fun XElement.originatingElement(): Element? {
```

Comment:
So far, I don't think we ever exposed JavaAP types from the API.
Maybe, it is better to have a more explicit name not to get confused with XElement? Maybe something like:

`originatingJavaAPElement`
or
`originatingElementForJavaPoet`
or
`originatingElementForPoet` to also cover for KotlinPoet?


might also make sense to move this to a real method in XElement interface so that if we ever have another type of it, we won't forget to implement this. But I'm fine with this implementation as well, just an idea.

Reply:
sure, I can move `originatingElementForPoet` to the interface

Reply:
actually I realized that publicly exposing this isn't actually that helpful, since if someone did want to use it for some custom writing they wouldn't be able to do so in KSP since they can't access the `KSFileAsOriginatingElement`. We would need to provide a more thought out abstraction for that, which seems overkill for now since it is unlikely someone is using neither javapoet or kotlinpoet, so I'll just go back to having this be internal